src/porepy/geometry/constrain_geometry.py

```diff
@@ -58,48 +72,58 @@ def lines_by_polygon(poly_pts, pts, edges):
                     int_pts = np.c_[int_pts, np.array(int_line.xy)]
-                    edges_kept.append(ei)
+                    edges_kept_aslist.append(ei)
 
     # define the list of edges
     int_edges = np.arange(int_pts.shape[1]).reshape((2, -1), order="F")
 
     # Also preserve tags, if any
-    if len(edges_kept) > 0:
-        edges_kept = np.array(edges_kept)
-        edges_kept.sort()
-        int_edges = np.vstack((int_edges, edges[2:, edges_kept]))
+    if len(edges_kept_aslist) > 0:
+        edges_kept_asarray = np.array(edges_kept_aslist)
+        edges_kept_asarray.sort()
+        int_edges = np.vstack((int_edges, edges[2:, edges_kept_asarray]))
+        edges_kept = np.array(edges_kept_asarray, dtype=int)
     else:
         # If no edges are kept, return an empty array with the right dimensions
         int_edges = np.empty((edges.shape[0], 0), dtype=int)
+        edges_kept = np.array(edges_kept_aslist, dtype=int)
 
-    return int_pts, int_edges, np.array(edges_kept, dtype=int)
+    return int_pts, int_edges, edges_kept
 
 
-def polygons_by_polyhedron(polygons, polyhedron, tol=1e-8):
-    """Constrain a sort of polygons in 3d to lie inside a, generally non-convex, polyhedron.
+def polygons_by_polyhedron(
+    polygons: Union[np.ndarray, list[np.ndarray]],
+    polyhedron: list[np.ndarray],
+    tol: float = 1e-8,
+) -> tuple[list[np.ndarray], np.ndarray]:
+    """Constrain polygons in 3d to lie inside a (generally non-convex) polyhedron.
 
-    Polygons not inside the polyhedron will be removed from descriptions.
-    For non-convex polyhedra, polygons can be split in several parts.
+    Polygons not inside the polyhedron will be removed from descriptions. For
+    non-convex polyhedra, polygons can be split in several parts.
 
     Parameters:
-        polygons (np.ndarray, or list of arrays): Each element is a 3xnum_vertex
-            array, describing the vertexes of a polygon.
-        polyhedron (list of np.ndarray): Each element is a 3 x num_vertex array,
-            describing the vertexes of the polygons that together form the
-            polygon
-        tol (double, optional): Tolerance used to compare points. Defaults to 1e-8.
+        polygons: Each element is an array of `shape=(3, num_vertex)`, describing the
+            vertexes of a polygon.
+        polyhedron: Each element is an array of `shape=(3, num_vertex)`, describing the
+            vertexes of the polygons that together form the polygon.
+        tol: Tolerance used to compare points.
 
     Returns:
-        list of np.ndarray: Of polygons lying inside the polyhedra.
-        np.ndarray: For each constrained polygon, corresponding list of its original
-            polygon
+        A tuple with two elements.
+
+        list of ndarray:
+            Polygons lying inside the polyhedra. Each array has
+            ``shape=(3, num_vertex)``.
+
+        ndarray ``(shape=(num_polygons, ), dytpe=int)``:
```

Comment:
Again maybe write len = num_polygons instead of (shape=(num_polygons, )?

Reply:
Same reason as the other comment.